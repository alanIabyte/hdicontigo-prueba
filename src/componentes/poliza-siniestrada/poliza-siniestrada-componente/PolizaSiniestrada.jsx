/* eslint-disable import/no-unresolved */
/* eslint-disable react/forbid-prop-types */
import React, { useState } from "react";
import PropTypes from "prop-types";
import { useHistory } from "react-router-dom";
import { useCookies } from "react-cookie";
import { useDispatch } from "react-redux";
import moment from "moment";
import "moment/locale/es-mx";
import IconoFlechaAbajo from "@material-ui/icons/KeyboardArrowDownRounded";
import IconoFlechaArriba from "@material-ui/icons/KeyboardArrowUpRounded";
import {
  EnvolvedorPolizaSiniestrada,
  ImagenCarro,
  TituloSeccion,
  NumeroPoliza,
  VigenciaPoliza,
  SeparadorSiniestros,
  ContenedorSiniestros,
  EncabezadoSiniestros,
  TextoSiniestros,
  NumeroSiniestros,
  ContenedorIconoFlecha,
  Siniestro,
  ImagenPortapapeles,
  FechaSiniestro,
  ContenedorBoton,
} from "./PolizaSiniestrada.styled";
import Constantes from "../../../recursos/constantes";
import carroDefault from "../../../recursos/iconos/hdi-c/mis-poliza/autos.png";
import IconoGrua from "../../../recursos/iconos/hdi-c/reporte-siniestro/grua.svg";
import IconoRobo from "../../../recursos/iconos/hdi-c/reporte-siniestro/robo.svg";
// import portapapeles from "../../../recursos/imagenes/portapapeles.png";
import IconPortaPapeles from "../../../recursos/iconos/hdi-c/detalle/gmm/1Info.svg";
import { BotonReportarSiniestro } from "../../poliza/poliza-componente/Poliza.styled";
import showConfig from "../../../utils/configs";

const diccionario = {
  poliza: "Póliza: ",
  vigencia: "Vigente hasta: ",
  BotonRegistraSin: "Reportar choque",
  siniestros: "Siniestro(s) activo(s): ",
};

const nombreCookie = Constantes.nombreDeCookie;

const PolizaSiniestrada = (props) => {
  const {
    enClickEnFoto,
    error,
    imagen,
    numeroPoliza,
    numeroPolizaCompuesto,
    siniestros,
    vehiculo,
    vigencia,
  } = props;

  const dispatch = useDispatch();
  const history = useHistory();
  const [mostrarSiniestros, asignarValorMostrarSiniestros] = useState(false);
  const [cookie] = useCookies([nombreCookie]);
  let objetoCookie = cookie[nombreCookie];

  const elegirSiguientePantalla = (siniestro) => {
    const {
      estatusReporte,
      numeroReporte,
      nombreReporta,
      ubicacion,
      tipoAtencion,
      idSubEvento,
    } = siniestro;
    dispatch({
      type: "AGREGAR",
      valor: {
        data: vehiculo,
      },
      indice: "datosVehiculo",
    });

    if (
      (estatusReporte === "A" || estatusReporte === "B") &&
      tipoAtencion === 0 &&
      idSubEvento !== 1 &&
      idSubEvento !== 21 &&
      idSubEvento !== 22
    ) {
      objetoCookie = { ...objetoCookie, nombreReporta };
      // establecerCookie(nombreCookie, JSON.stringify(objetoCookie), {
      //   path: "/",
      // });
      const { latitud, longitud } = ubicacion;
      if (latitud && longitud) {
        dispatch({
          type: "AGREGAR",
          valor: { lat: latitud, lng: longitud },
          indice: "coordenadasIniciales",
        });
      }
      history.push({
        pathname: "menu-espera",
        search: `?numeroReporte=${numeroReporte}`,
      });
    } else if (idSubEvento === 1) {
      /**
       * 1 => Reporte de grúa
       */
      dispatch({
        type: "AGREGAR",
        valor: numeroReporte,
        indice: "numeroReporteGrua",
      });
      history.push({
        pathname: "/pasos-progreso-grua",
      });
    } else if (idSubEvento === 21 || idSubEvento === 22) {
      /**
       * Cuando es de robo
       * 21 => Robo con violencia
       * 22 => Robo estacionado
       */
      dispatch({
        type: "AGREGAR",
        indice: "datosReporteRobo",
        valor: {
          numeroPoliza: numeroPolizaCompuesto,
        },
      });

      dispatch({
        type: "AGREGAR",
        valor: estatusReporte,
        indice: "estadoReporte",
      });
      dispatch({
        type: "AGREGAR",
        valor: siniestro.numeroSiniestro,
        indice: "numeroSiniestroReporte",
      });
      if (showConfig.showRobo) {
        // !Se cambia la redirección de pasos-progreso a menu-espera-robo ya que ahí se hará la validación para redirigir a pasos progreso
        history.push({
          pathname: "menu-espera-robo",
          search: `?numeroReporte=${numeroReporte}`,
        });
      } else {
        history.push({
          pathname: "pasos-progreso",
          search: `?numeroReporte=${numeroReporte}`,
        });
      }
    } else {
      dispatch({
        type: "AGREGAR",
        valor: estatusReporte,
        indice: "estadoReporte",
      });
      dispatch({
        type: "AGREGAR",
        valor: siniestro.numeroSiniestro,
        indice: "numeroSiniestroReporte",
      });
      history.push({
        pathname: "pasos-progreso",
        search: `?numeroReporte=${numeroReporte}`,
      });
    }
  };

  const validaIcono = (idSubEvento) => {
    switch (idSubEvento) {
      case 1:
        return IconoGrua;
      case 21:
      case 22:
        return IconoRobo;
      default:
        return IconPortaPapeles;
    }
  };
  return (
    <EnvolvedorPolizaSiniestrada>
      <ImagenCarro
        src={imagen || carroDefault}
        onClick={enClickEnFoto}
        data-numero-poliza={numeroPoliza}
        error={error}
      />
      <TituloSeccion>{vehiculo}</TituloSeccion>
      <NumeroPoliza>
        {diccionario.poliza} {numeroPoliza}
      </NumeroPoliza>
      <VigenciaPoliza>
        {diccionario.vigencia} {moment(vigencia).format("DD/MMM/YY")}
        {/* <BotonRegistraSin
          onClick={() => {
            history.push({
              pathname: "/ingreso-poliza",
              search: `?numeroPoliza=${numeroPolizaCompuesto}&vin=${vin}`,
            });
          }}
        >
          {diccionario.BotonRegistraSin}
        </BotonRegistraSin> */}
      </VigenciaPoliza>

      <SeparadorSiniestros />
      <ContenedorSiniestros>
        <EncabezadoSiniestros
          onClick={() => {
            asignarValorMostrarSiniestros(!mostrarSiniestros);
          }}
        >
          <TextoSiniestros>{diccionario.siniestros}</TextoSiniestros>
          <NumeroSiniestros>{siniestros.length}</NumeroSiniestros>
          <ContenedorIconoFlecha>
            {mostrarSiniestros ? <IconoFlechaArriba /> : <IconoFlechaAbajo />}
          </ContenedorIconoFlecha>
        </EncabezadoSiniestros>
        {mostrarSiniestros &&
          siniestros.map((valor, indice) => (
            <Siniestro key={valor.numeroReporte}>
              <ImagenPortapapeles
                src={validaIcono(siniestros[indice].idSubEvento)}
              />
              {siniestros[indice].numeroSiniestro ? (
                <TituloSeccion>
                  {siniestros[indice].numeroSiniestro}
                </TituloSeccion>
              ) : (
                <TituloSeccion>
                  {siniestros[indice].numeroReporte}
                </TituloSeccion>
              )}
              <ContenedorBoton>
                <BotonReportarSiniestro
                  reporte={false}
                  onClick={() => {
                    elegirSiguientePantalla(siniestros[indice]);
                  }}
                  deshabilitado={false}
                >
                  {/* <IconoInfoPagos className="ico" width={22} style={{color: "white"}}/> */}
                  Seleccionar
                </BotonReportarSiniestro>
              </ContenedorBoton>
              <FechaSiniestro>
                {moment(siniestros[indice].fechaReporte).format("DD/MMM/YY")}
              </FechaSiniestro>
            </Siniestro>
          ))}
      </ContenedorSiniestros>
    </EnvolvedorPolizaSiniestrada>
  );
};

PolizaSiniestrada.defaultProps = {
  vehiculo: "#Nombre o modelo de auto YYYY",
  numeroPoliza: "XXX-XX-XXX",
  numeroPolizaCompuesto: "x-x-x",
  vigencia: "DD/MM/YYY",
  siniestros: [
    {
      numeroSiniestro: "000000",
      numeroReporte: "321832",
      fechaReporte: "01/Ene/2021",
    },
  ],
  enClickEnFoto: () => {},
  imagen: "",
  error: false,
};

PolizaSiniestrada.propTypes = {
  enClickEnFoto: PropTypes.func,
  error: PropTypes.bool,
  imagen: PropTypes.string,
  numeroPoliza: PropTypes.string,
  numeroPolizaCompuesto: PropTypes.string,
  siniestros: PropTypes.arrayOf(PropTypes.object),
  vehiculo: PropTypes.string,
  vigencia: PropTypes.string,
};

export default PolizaSiniestrada;
