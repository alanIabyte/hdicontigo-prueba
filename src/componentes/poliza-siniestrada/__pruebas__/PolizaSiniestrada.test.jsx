import React from "react";
import renderer from "react-test-renderer";
import PolizaSiniestrada from "../poliza-siniestrada-componente/PolizaSiniestrada";

jest.mock("react-redux", () => ({
  useDispatch: jest.fn().mockReturnValue({}),
}));

describe("#PolizaSiniestrada", () => {
  it("se despliega correctamente", () => {
    const tree = renderer.create(<PolizaSiniestrada />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
