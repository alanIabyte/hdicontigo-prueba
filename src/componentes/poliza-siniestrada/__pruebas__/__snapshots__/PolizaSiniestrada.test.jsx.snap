// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#PolizaSiniestrada se despliega correctamente 1`] = `
<div
  className="sc-ikkxIA oBEQp"
>
  <img
    className="sc-dAbbOL cjYUcy"
    data-numero-poliza="XXX-XX-XXX"
    onClick={[Function]}
    src="test-file-stub"
  />
  <div
    className="sc-iGgWBj sc-feUZmu fgQoGq ekOaQc"
  >
    #Nombre o modelo de auto YYYY
  </div>
  <div
    className="sc-iGgWBj sc-fUnMCh fgQoGq"
  >
    Póliza: 
     
    XXX-XX-XXX
  </div>
  <div
    className="sc-eldPxv sc-hzhJZQ jeNqaZ jtGwLX"
  >
    Vigente hasta: 
     
    Fecha inválida
  </div>
  <div
    className="sc-hmdomO eRTxhy"
  />
  <div
    className="sc-bXCLTC jsrxpX"
  >
    <div
      className="sc-jsJBEP jWWLCa"
      onClick={[Function]}
    >
      <div
        className="sc-kpDqfm sc-eeDRCY bTCkFB dncNPu"
      >
        Siniestro(s) activo(s): 
      </div>
      <div
        className="sc-kpDqfm sc-koXPp bTCkFB jVjrit"
      >
        1
      </div>
      <div
        className="sc-bmzYkS eeHdGg"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.12 9.29L12 13.17l3.88-3.88c.39-.39 1.02-.39 1.41 0 .39.39.39 1.02 0 1.41l-4.59 4.59c-.39.39-1.02.39-1.41 0L6.7 10.7a.9959.9959 0 010-1.41c.39-.38 1.03-.39 1.42 0z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
